go/parquet/file/column_writer_types.gen.go.tmpl

```diff
@@ -137,13 +137,13 @@ func (w *{{.Name}}ColumnChunkWriter) WriteBatchSpaced(values []{{.name}}, defLev
 
     w.writeLevelsSpaced(batch, levelSliceOrNil(defLevels, offset, batch), levelSliceOrNil(repLevels, offset, batch))
     if values != nil {
-      vals = values[valueOffset:]
+      vals = values[valueOffset:][:info.numSpaced()]
```

Comment:
why was this needed? also, if you're going to add this it should be `vals = values[valueOffset : info.numSpaced()]` no need for the extra set of brackets.

Reply:
I made the change to use the following form: `vals = values[valueOffset:valueOffset+info.numSpaced()]`